### ISSUE TITLE
Default event space bijector for discrete variables

### ISSUE THREAD

**Issue by chrism0dwk** [tensorflow/probability#1407]
Hi,

I was wondering why the semantics for `experimental_default_event_space_bijector`  returns `None` for distributions that don't have an explicit bijector.  For example
```python
foo = tfd.Binomial(total_count=10, probs=0.5)
foo.experimental_default_event_space_bijector()  # => None
```

Wouldn't it be more useful to return `tfb.Identity` for such distributions so as not to break `tfd.JointDistribution*.experimental_default_event_space_bijector()`?  The concept of a default bijector is a property of the TFP library and not the underlying probability distribution (e.g. devs have chosen `tfb.SoftMax` instead of `tfb.Exp` for a bounded-continuous distribution when either is mathematically valid), so this wouldn't violate any laws of probability whilst facilitating further automation of MCMC building...

**Comment by brianwa84** [comment]
You can get a joint bijector for the non-discrete parts of the state space by using `jd.experimental_pin(foo=obsv).experimental_default_event_space_bijector()`.

**Comment by brianwa84** [comment]
Toward your second comment: the default bijector is custom for each distribution with the idea that the bijector maps from unconstrained R^n to the support of the distribution in question (and back). Since there is no two-way mapping from reals to ints, we don't have support bijectors for discrete distributions.

**Comment by chrism0dwk** [comment]
@brianwa84 That totally makes sense mathematically, and I guess isn't a problem in the continuous-everywhere world!  You know me, I like to mix things up with discrete spaces now and then ;-)

Thanks for the advice ^^

**Comment by brianwa84** [comment]
Actually if you want something a little more invasive, you could explicitly
instantiate a _DefaultJointBijector and provide the bijector_fn argument.
We could maybe add a path for funneling that arg through if it's useful.
https://cs.opensource.google/tensorflow/probability/+/main:tensorflow_probability/python/distributions/joint_distribution.py;l=1101

It does look like we fall back to Identity in _conditioning_bijectors here
https://cs.opensource.google/tensorflow/probability/+/main:tensorflow_probability/python/distributions/joint_distribution.py;l=1139
but not in __init__.

I wonder if that's an oversight?

**Comment by chrism0dwk** [comment]
So you're suggesting something like:
```python
foo  = tfd.Binomial(total_count=100, probs=0.5, default_bijector=tfb.Identity())
```
for which the default argument value would be `default_bijector=None`?

That would certainly work for me, *and* would have the advantage of making the default choice of bijector very explicit to the user.  So for example, the signature for `tfd.Gamma.__init__` would become
```python
class Gamma(tfd.Distribution):
    def __init__( concentration,
               rate=None,
               log_rate=None,
               validate_args=False,
               allow_nan_stats=True,
               force_probs_to_zero_outside_support=False,
               default_bijector=tfb.Softmax(),
               name='Gamma'):
    pass
```
allowing the user to use `tfb.Exp` if they preferred.

**Comment by chrism0dwk** [comment]
> We could maybe add a path for funneling that arg through if it's useful.

That would definitely be useful -- actually looking at the _DefaultJointBijector function, would a reasonable default be to detect when a component distribution's `experimental_default_event_space_bijector` is `None` and substitute in `tfb.Identity()`, maybe with a warning to the user?
